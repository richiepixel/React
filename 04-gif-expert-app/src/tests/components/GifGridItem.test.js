import React from 'react';
import { shallow } from 'enzyme';
import GifGridItem from '../../Components/GifGridItem';

const img = {
  title: 'legend of zelda GIF',
  url:
    'https://media2.giphy.com/media/5MjQ87rBCRIB2/giphy.gif?cid=52d5781amo7m364mwu12b47its9vkdcwnpqx9gndzfykpgyr&rid=giphy.gif',
};

describe('Pruebas en <GifGridItem />', () => {
  test('debe de mostrar el componente correctamente', () => {
    const wrapper = shallow(<GifGridItem {...img} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('debe de tener un pÃ¡rrafo con el title', () => {
    const wrapper = shallow(<GifGridItem title={img.title} url={img.url} />);
    const p = wrapper.find('p');
    expect(p.text().trim()).toBe(img.title);
  });

  test('debe de tener la imagen igual al url y alt de los props', () => {
    const wrapper = shallow(<GifGridItem {...img} />);
    const imgTag = wrapper.find('img');

    expect(imgTag.props().src).toBe(img.url);
    expect(imgTag.prop('alt')).toBe(img.title);
  });

  test('debe de tener la className animate__fadeInDown', () => {
    const wrapper = shallow(<GifGridItem title={img.title} url={img.url} />);
    const div = wrapper.find('div');
    expect(div.is('.animate__fadeInDown')).toBeTruthy();
  });
});
