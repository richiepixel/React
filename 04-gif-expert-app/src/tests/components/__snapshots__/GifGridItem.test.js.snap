// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeInDown"
>
  <img
    alt="legend of zelda GIF"
    src="https://media2.giphy.com/media/5MjQ87rBCRIB2/giphy.gif?cid=52d5781amo7m364mwu12b47its9vkdcwnpqx9gndzfykpgyr&rid=giphy.gif"
  />
  <p>
    legend of zelda GIF
  </p>
</div>
`;
